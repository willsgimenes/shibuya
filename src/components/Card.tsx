import React, { FC, ReactNode } from 'react';
import styled from 'styled-components';

const StyledCard = styled.div.attrs(() => ({
  className: 'bg-white rounded-md shadow',
}))``;

export interface CardProps {
  children: ReactNode;
}

export const Card: FC<CardProps> = ({ children, ...props }) => (
  <StyledCard {...props}>{children}</StyledCard>
);
