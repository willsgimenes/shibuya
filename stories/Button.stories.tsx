import React from 'react';
import { Meta, Story } from "@storybook/react";
import { Button, ButtonProps } from '../src/components/Button';

const meta: Meta = {
    title: 'Button',
    component: Button,
    argTypes: {
        text: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expanded: true }
    }
}

export default meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const PrimaryButton = Template.bind({});
PrimaryButton.args = { text: 'Button', variant: 'primary' };

export const SecondaryButton = Template.bind({});
SecondaryButton.args = { text: 'Button', variant: 'secondary' }

export const TertitaryButton = Template.bind({});
TertitaryButton.args = { text: 'Button', variant: 'tertiary' }
