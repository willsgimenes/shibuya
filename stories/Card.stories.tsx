import React from "react";
import { Meta, Story } from "@storybook/react";
import { Card, CardProps } from "../src/components/Card";

const meta: Meta = {
    title: 'Card',
    component: Card,
}

export default meta;

const content = <div className='p-4'><p>渋谷区</p></div>

const Template: Story<CardProps> = (args) => <Card {...args} />

export const DefaultCard = Template.bind({});

DefaultCard.args = { children: content }
