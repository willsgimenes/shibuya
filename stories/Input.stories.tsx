import React, { useState, useRef } from 'react';
import { Meta, Story } from "@storybook/react";
import { Input, InputProps } from '../src/components/Input';

const meta: Meta = {
    title: 'Input',
    component: Input,
    argTypes: {
        value: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expdanded: true }
    }
}

export default meta;

const Template: Story<InputProps> = (args) => React.createElement(() => {
    const [state, setState] = useState(args.value)
    const inputRef = useRef();

    return <Input
        innerRef={inputRef}
        value={state}
        label={args.label}
        onChange={({ target: { value } }) => setState(value)}
        placeholder={args.placeholder}
        disabled={args.disabled}
    />
})

export const DefaultInput = Template.bind({});
DefaultInput.args = { value: 'Yokohama Blues', placeholder: 'Insert text', disabled: false, label: 'song' }
