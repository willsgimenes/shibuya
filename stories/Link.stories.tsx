import React from "react";
import { Meta, Story } from "@storybook/react";
import { Link as SCLink, LinkProps } from '../src/components/Link';

const meta: Meta = {
    title: 'Link',
    component: SCLink,
}

export default meta;

const Template: Story<LinkProps> = (args) => (
    <SCLink {...args}>{args.children}</SCLink>
)

export const Link = Template.bind({});
Link.args = {
    url: "https://open.spotify.com/track/1EJOWPgFH4ThdIFdvbKA2B?si=Zdo4k4UbRnShLeKTHtDr6Q",
    external: true,
    children: 'Listen Yokohama blues',
}