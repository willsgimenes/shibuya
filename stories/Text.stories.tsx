import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Text, TextProps } from "../src/components/Text";

const meta: Meta = {
    title: 'Typography',
    component: Text,
}

export default meta;

const Template: Story<TextProps> = () => <div>
    <Text variant="h1">Heading 1</Text>
    <Text variant="h2">Heading 2</Text>
    <Text variant="h3">Heading 3</Text>
    <Text variant="h4">Heading 4</Text>
    <Text variant="h5">Heading 5</Text>
    <Text variant="p">texto</Text>
    <Text variant='small'>small text</Text>
</div>

export const DefaultTypography = Template.bind({})