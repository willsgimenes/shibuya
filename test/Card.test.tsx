import * as React from 'react';
import { render } from '@testing-library/react';
import { DefaultCard } from '../stories/Card.stories';

describe('Card', () => {
  it('Should render the card without crashing', () => {
    const { getByText } = render(
      <DefaultCard>
        <div>渋谷区</div>
      </DefaultCard>
    );

    expect(getByText('渋谷区')).toBeInTheDocument();
    expect(
      <DefaultCard>
        <div>渋谷区</div>
      </DefaultCard>
    ).toMatchSnapshot();
  });
});
