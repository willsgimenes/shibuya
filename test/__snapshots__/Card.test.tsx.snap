// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render the card without crashing 1`] = `
<bound Template>
  <div>
    渋谷区
  </div>
</bound Template>
`;
